backend/src/prisma/recipes.js

```diff
@@ -1,29 +1,123 @@
-const { PrismaClient } = require('@prisma/client')
+const Prisma = require("prisma");
+const { DeleteRecipeError } = require('../errors');
+const { PrismaClient } = require('@prisma/client');
 
 const prisma = new PrismaClient()
 
 
 async function getRecipes() {
-    return await prisma.recipe.findMany({
-        include: {
-            RecipeIngredient: true,
-        },
-    });
+    let recipes = await prisma.recipe.findMany();
+    for (let i = 0; i < recipes.length; i++) {
+        let recipeIngredients = await prisma.recipeIngredient.findMany({
+            where: {
+                recipeId: recipes[i].id,
+            }
+        });
+        let ingredientList = [];
+        let ing;
+        for (let j = 0; j < recipeIngredients.length; j++) {
+            ing = await prisma.ingredient.findUnique({
+                where: {
+                    id: recipeIngredients[j].ingredientId,
+                }
+            });
+            ing.amount = recipeIngredients[j].amount;
+            ingredientList.push(ing);
+        }
+        recipes[i].ingredients = ingredientList;
+    }
+    return recipes;
 }
 
-
-async function createRecipe() {
-    return null
+async function createRecipe(recipe) {
+    let createdRecipe = await prisma.recipe.create({
+        data: {
+            name: recipe.name,
+            description: recipe.description,
+            preparation_length: recipe.preparationLength,
+            final_amount: recipe.ingredients.length
+        },
+    });
+    for (let i = 0; i < recipe.ingredients.length; i++) {
```

Comment:
Take a look at https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-writes 
"Provide **transactional guarantees** for creating, updating or deleting data across multiple tables in a single Prisma Client query", that sounds nice.

Reply:
todo with anca